docs/manual/scala/guide/broker/code/docs/scaladsl/mb/PublishServiceSpec.scala

```diff
@@ -0,0 +1,81 @@
+package docs.scaladsl.mb
+
+import com.lightbend.lagom.scaladsl.api.broker.Topic
+import com.lightbend.lagom.scaladsl.server.{LagomApplication, LagomApplicationContext, LagomServer, LocalServiceLocator}
+import com.lightbend.lagom.scaladsl.testkit.{ServiceTest, TestTopicComponents}
+import play.api.libs.ws.ahc.AhcWSComponents
+import org.scalatest.{AsyncWordSpec, Matchers}
+import akka.{NotUsed, Done}
+import akka.stream.scaladsl.Source
+import akka.stream.testkit.scaladsl.TestSink
+import akka.stream.testkit.TestSubscriber
+import akka.stream.testkit.TestSubscriber.Probe
+
+abstract class PublishApplication(context: LagomApplicationContext)
+  extends LagomApplication(context)
+    with TestTopicComponents
```

Comment:
You wouldn't usually mix in `TestTopicComponents` here, since it's not what you want when you run in production.  Instead, you probably wouldn't mix in any topic factory here, rather, in a sub type of this, you would mix in `KafkaTopicComponents`, and then in your tests, you would mix in `TestTopicComponents`.  Of course, this code snippet isn't shown in the documentation, but basically what I'm saying is move mixing in this trait down to the test.